### DISCUSSION TITLE
Isaac Z. Schlueter - npm

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#219] [Announcements]
- https://github.com/isaacs
- https://www.tier.run
- https://node-tap.org
- https://fosstodon.org/@isaacs
- https://twitter.com/izs

## Topics

- npm 
- open source
- [https://node-tap.org](https://t.co/hArWXgEjph), lots to dig into there about typescript, esm, cjs, testing theory in general, etc.

### Past

- [x] What was it like working working on node when it came out? Did you ever think it would become what it is today
- [x] npm and node became managed in very different ways. Do you think npm breaking away as a company was better for the long term health of the platform?
- [x] A lot of newsworthy events have happened at npm over the years. What is the most memorable "war story" you have?

#### Tier

- [x] What is tier? What developer pain point is it meant to solve?
- [ ] ~What are [priceops](https://priceops.org)? (@zephraph I feel you'd better craft this question)~

#### Node

- [x] Node's transition from cjs to esm has been compared to python2 to 3. Why do you think the transition has take so long and what do you think needs to happen to get it over the finish line? Is there anything that could have been done differently?
- [x] You make lots of tiny npm packages. Have you created and cool ones lately? I personally like [tshy](https://github.com/isaacs/tshy)

#### Tap

- [ ] What is tap?
- [ ] ~What makes tap different from other testing libraries?~
- [ ] ~How does the typescript support work with all the plugins? Was it tricky to get right?~
- [ ] ~What's your favorite feature of tap? (there's a lot) Code coverage seems like an interesting topic.~
- [ ] ~Is the plugin system itself a package, seems cool!~

### Future

- [x] After having working on such an influential platform have your opinions on distributing software and OSS changed?
- [ ] What do you think the future of javascript is? Will new runtimes take over? Or do you think node/npm will be here for a bit longer?

**Comment by hipstersmoothie** [comment]
- https://rivet.ironcladapp.com/
- https://jsx.email/

**Comment by just-be-dev** [comment]
- https://github.com/aidenybai/pattycake
- https://github.com/TodePond/DreamBerd

**Comment by hipstersmoothie** [comment]
https://neovim.io

**Comment by isaacs** [comment]
Re the "Future" question about js runtimes: https://twitter.com/izs/status/1704063803695173632

tooltip: neovim and typescript

[forwardemail.net](https://forwardemail.net/en)
